hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorServiceProxy.java

```diff
@@ -440,9 +439,8 @@ private String extractNameOrGenerateOne(Object command) {
     }
 
     private void attachHazelcastInstance(Object object) {
-        if (object instanceof HazelcastInstanceAware) {
-            ((HazelcastInstanceAware) object).setHazelcastInstance(getNodeEngine().getHazelcastInstance());
-        }
+        getNodeEngine().getSerializationService()
+                       .getManagedContext().initialize(object);
```

Comment:
I understand [`getManagedContext` may return a `null` context?](https://github.com/vbekiaris/hazelcast/blob/4de7f846e0dd85354de7af94bd7ff81ac2487c24/hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java#L186-L188) Also, since this method now does more than just setting the `HazelcastInstance` on a `HazelcastInstanceAware` object, maybe method should be renamed?

Reply:
Regarding the null, i might have missed sth, but was looking at the way we initialize it, and it seems that we always have a HazelcastManagedContext, which includes within it, the external context (the one from the Config, eg. Spring). The external maybe null, but the wrapper is always there. see. https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceImpl.java#L122

About the naming, indeed its not just setting the instance, I only left it like this due for convenience, since its adhering to the HazelcastInstanceAware API, allowing me to get the instance from the Proxy or after deserialisation. WDYT ?

Reply:
Yes, looks like it's never `null`, so it's rather a superfluous null-check on the `AbstractSerializationService` side.